src/vecs/client.py

```diff
@@ -65,6 +66,21 @@ def __init__(self, connection_string):
             with sess.begin():
                 sess.execute(text("create schema if not exists vecs;"))
                 sess.execute(text("create extension if not exists vector;"))
+                self.vector_version: str = sess.execute(
+                    text(
+                        "select installed_version from pg_available_extensions where name = 'vector' limit 1;"
+                    )
+                ).scalar_one()
+
+        if not self._supports_hnsw():
+            warnings.warn(
+                UserWarning(
+                    f"vecs will drop support for pgvector < 0.5.0 in version 1.0. Consider updating to latest postgres"
+                )
+            )
+
+    def _supports_hnsw(self):
+        return not self.vector_version.startswith("0.4")
```

Comment:
is it ok to ignore 0.3 and below?

Reply:
good call, added in case others have older versions deployed